builtin/rebase--interactive.c

```diff
@@ -224,6 +226,15 @@ int cmd_rebase__interactive(int argc, const char **argv, const char *prefix)
 		warning(_("--[no-]rebase-cousins has no effect without "
 			  "--rebase-merges"));
 
+	if (opts.ignore_whitespace) {
+		struct strbuf buf = STRBUF_INIT;
+		if (raw_strategies)
+			strbuf_addstr(&buf, raw_strategies);
+		strbuf_addstr(&buf, " ignore-space-change");
+		raw_strategies = xstrdup(buf.buf);
```

Comment:
If we are using `xstrdup` should we release the memory afterwards? I noticed we are not doing it for other vars too. (e.g. rebase_options->strategy).

Reply:
In general, yes, we need to make sure we free memory we allocate.  If it's a widespread leak within the file in question, it sometimes makes sense to first do a fixup commit (or commits) that fixes the other problems currently present before adding your changes.  And sometimes, it just makes sense to not worry about it right now and just note that someone needs to come in and fix up all the callers in that file.

I know that doesn't directly answer your question, but it's gray area.  It's the kind of thing I'd call out in my 'cover letter' when submitting patches to the list and ask folks for their opinion.  That alone will sometimes spur someone else to do all the cleanup.  :-)

Reply:
There is a function call inside xstrdup's definition which does nothing? Why is so?

Reply:
I don't understand your question.

Reply:
Sorry, I should have been clearer. I am talking about [this](https://github.com/git/git/blob/master/wrapper.c#L46) function [`try_to_free_routine`](https://github.com/git/git/blob/master/wrapper.c#L11) which points to [`do_nothing`](https://github.com/git/git/blob/74583d89127e21255c12dd3c8a3bf60b497d7d03/wrapper.c#L7-L9) that actually does nothing! Why do we have such a funtion?

Reply:
It's not always true that it points to `do_nothing()` though.  If you just `git grep` for `try_to_free_routine`, this result stands out in particular:

```
builtin/pack-objects.c: old_try_to_free_routine = set_try_to_free_routine(try_to_free_from_threads);
```

Even better you can dig into the history of why this was introduced in the first place, `git blame` is your friend here.  That leads to a9a746364bd26d333c7229c6f7e851b507cd284a, but there are probably other interesting commits, such as why `do_nothing` was introduced, which has happened in a separate commit, or what `release_pack_memory` did, that you can find out about by just browsing through Gits history if you want to.

This is one of the big benefits the attention to detail we're paying in commit messages has.